kasenvr/project-athena
# Zone occlusion

All entities have a new property `renderWithZones`, a vector of Uuids.
An entity will only render if:
1. it does not specify this property
2. it does specify the property and your avatar is currently inside one of the zones indicated **(currently, the zone also has to be marked as visible.  is that desirable?)**

**This PR does not expose the new property to Create.  There's not currently any other list property like this that's exposed and I'm not sure how best to do it.**

This PR also does not handle physics...although the entities stop rendering, they still affect you physically.  I don't see any way around this...now, two people can be seeing different things, but they need to agree on the physics of everything.

Test plan:
- Go somewhere with no zones and run [this script](https://gist.githubusercontent.com/HifiExperiments/24cbe3779d20ed5cc30966a9aa16fe16/raw/10731f9d69b4555c5571a1b62aadde79ed2e2cfc/ZoneOcclusionTest.js).  You'll see:
![image](https://user-images.githubusercontent.com/53453710/72215128-c107c380-34c3-11ea-836b-592c27b25a2b.png)
- Press "o".  You'll see:
![image](https://user-images.githubusercontent.com/53453710/72215132-c95ffe80-34c3-11ea-97df-ca45a54d37cb.png)
- Press "o".  You'll see:
![image](https://user-images.githubusercontent.com/53453710/72215137-d11fa300-34c3-11ea-8ec3-975615c6fb8c.png)
- Press "o".  You'll see:
![image](https://user-images.githubusercontent.com/53453710/72215139-d846b100-34c3-11ea-8c38-cde2a0b101e5.png)
- Press "i" and go through all the zones by pressing "o" multiple times again.  The entities will change to draw in front, but otherwise you'll see the same things.
